sdk/test/Services/S3/UnitTests/Custom/ResponseMapperTests.cs

```diff
@@ -209,6 +216,491 @@ public void MapPutObjectResponse_AllMappedProperties_WorkCorrectly()
+                var sourceProperty = sourceType.GetProperty(resolvedPropertyName);
+                if (sourceProperty?.CanWrite == true)
+                {
+                    var testValue = GenerateTestValue(sourceProperty.PropertyType, propertyName);
+                    sourceProperty.SetValue(sourceRequest, testValue);
+                    testDataValues[propertyName] = testValue;
+                }
+            }
+
+            // Add inherited properties for comprehensive testing
+            sourceRequest.InputStream = new MemoryStream(1024);
+
+            var multipartUploadCommand = new MultipartUploadCommand(null, null, sourceRequest);
+
+            // Map the response
+            var initiateResponse = new InitiateMultipartUploadResponse
+            {
+                UploadId = "test-upload-id"
+            };
+            var mappedRequest = multipartUploadCommand.ConstructUploadPartRequest(1, 1024, initiateResponse);
+            Assert.IsNotNull(mappedRequest, "Mapped request should not be null");
+
+            // Verify all mapped properties using reflection
+            var targetType = typeof(UploadPartRequest);
+            var failedAssertions = new List<string>();
+
+            foreach (var propertyName in uploadPartRequestMappings)
+            {
+                // Resolve alias to actual property name for reflection lookups
+                var resolvedSourcePropertyName = ResolvePropertyName(propertyName, sourceType.Name);
+                var resolvedTargetPropertyName = ResolvePropertyName(propertyName, targetType.Name);
+                var sourceProperty = sourceType.GetProperty(resolvedSourcePropertyName);
+                var targetProperty = targetType.GetProperty(resolvedTargetPropertyName);
+
+                if (sourceProperty == null)
+                {
+                    failedAssertions.Add($"Source property '{propertyName}' (resolved to: {resolvedSourcePropertyName}) not found in {sourceType.Name}");
+                    continue;
+                }
+
+                if (targetProperty == null)
+                {
+                    failedAssertions.Add($"Target property '{propertyName}' (resolved to: {resolvedTargetPropertyName}) not found in {targetType.Name}");
+                    continue;
+                }
+
+                var sourceValue = sourceProperty.GetValue(sourceRequest);
+                var targetValue = targetProperty.GetValue(mappedRequest);
+
+                // Special handling for complex object comparisons
+                if (!AreValuesEqual(sourceValue, targetValue))
+                {
+                    failedAssertions.Add($"{propertyName}: Expected '{sourceValue ?? "null"}', got '{targetValue ?? "null"}'");
+                }
+            }
+
+            // Report any failures
+            if (failedAssertions.Any())
+            {
+                Assert.Fail($"Property mapping failures:\n{string.Join("\n", failedAssertions)}");
+            }
+        }
```

Comment:
nit: can you make a helper function for this code. its re-used in all of the test cases.

Reply:
Fixed